### PR TITLE
Save up to 5 conference aliases in shared preferences

### PR DESCRIPTION
**Pull request by azduda**
- Save up to 5 last conference aliases, without reduplicate. 
- On start, conference alias field is populated with latest saved alias.
- To select from last saved, click the storage icon next to the clear icon in the text field 'conference alias' and select a value from the list.

### CODE REVIEW

File: test_app/lib/screens/join_screen.dart [hunk 1]
```diff
@@ -154,13 +157,17 @@ class _JoinConferenceContentState extends State<JoinConferenceContent> {
...
+                    autovalidateMode: AutovalidateMode.onUserInteraction,
+                    child: InputTextFormField(
+                        labelText: 'Conference alias',
+                        controller: conferenceAliasTextController,
+                        focusColor: Colors.deepPurple,
+                        isStorageNeeded: true,
+                        onStorageIconTap: () async {
+                          await showAliasSelectorDialog(context,
+                              SharedPreferencesHelper().conferenceAliases);
+                        })),
```
Comment (kdrob-dolby) [inline test_app/lib/screens/join_screen.dart:170]: Let's use trailing commas for proper formatting.

```suggestion
                    key: formKeyAlias,
                  autovalidateMode: AutovalidateMode.onUserInteraction,
                  child: InputTextFormField(
                    labelText: 'Conference alias',
                    controller: conferenceAliasTextController,
                    focusColor: Colors.deepPurple,
                    isStorageNeeded: true,
                    onStorageIconTap: () async {
                      await showAliasSelectorDialog(
                        context,
                        SharedPreferencesHelper().conferenceAliases,
                      );
                    },
                  ),
                ),
```

File: test_app/lib/screens/join_screen.dart [hunk 2]
```diff
@@ -486,6 +494,61 @@ class _JoinConferenceContentState extends State<JoinConferenceContent> {
...
+                ]),
+              )),
+              actions: [
+                TextButton(
+                    style: TextButton.styleFrom(primary: Colors.deepPurple),
+                    onPressed: () => Navigator.of(context).pop(),
+                    child: const Text('Cancel')),
+              ]);
+        });
+  }
```
Comment (kdrob-dolby) [inline test_app/lib/screens/join_screen.dart:532]: Formatting

```suggestion
  Future<void> showAliasSelectorDialog(
    BuildContext context,
    List<String>? conferenceAliases,
  ) async {
    return showDialog<void>(
        context: context,
        builder: (BuildContext context) {
          return AlertDialog(
              title: const Text('Choose from recently saved'),
              actionsOverflowButtonSpacing: 10,
              content: SingleChildScrollView(
                child: SizedBox(
                  width: double.maxFinite,
                  child: Column(
                    mainAxisSize: MainAxisSize.min,
                    children: [
                      if (conferenceAliases != null)
                        ListView.builder(
                          shrinkWrap: true,
                          itemCount: conferenceAliases.length,
                          itemBuilder: (context, index) {
                            String alias = conferenceAliases[index];
                            return ListTile(
                              title: Text(alias),
                              onTap: () => onAliasTap(alias),
                            );
                          },
                        )
                      else
                        const Align(
                          alignment: Alignment.centerLeft,
                          child: Text('No aliases in storage.'),
                        )
                    ],
                  ),
                ),
              ),
              actions: [
                TextButton(
                  style: TextButton.styleFrom(primary: Colors.deepPurple),
                  onPressed: () => Navigator.of(context).pop(),
                  child: const Text('Cancel'),
                ),
              ]);
        });
  }
```

File: test_app/lib/widgets/text_form_field.dart [hunk 3]
```diff
@@ -51,9 +55,20 @@ class _InputTextFormFieldState extends State<InputTextFormField> {
...
+              widget.isStorageNeeded
+                  ? IconButton(
+                      onPressed: widget.onStorageIconTap,
+                      icon: const Icon(Icons.storage),
+                      color: Colors.grey)
+                  : const SizedBox.shrink(),
+              IconButton(
+                  onPressed: widget.controller!.clear,
+                  icon: const Icon(Icons.clear, color: Colors.grey))
+            ]),
```
Comment (kdrob-dolby) [inline test_app/lib/widgets/text_form_field.dart:71]: Formatting
```suggestion
        suffixIcon: Row(
          mainAxisAlignment: MainAxisAlignment.end,
          mainAxisSize: MainAxisSize.min,
          children: [
            widget.isStorageNeeded
                ? IconButton(
                    onPressed: widget.onStorageIconTap,
                    icon: const Icon(Icons.storage),
                    color: Colors.grey,
                  )
                : const SizedBox.shrink(),
            IconButton(
              onPressed: widget.controller!.clear,
              icon: const Icon(Icons.clear, color: Colors.grey),
            )
          ],
        ),
```

**Review by kdrob-dolby — CHANGES_REQUESTED** [review]

File: test_app/lib/widgets/text_form_field.dart (hunk 3 above)
Comment (azduda) [reply test_app/lib/widgets/text_form_field.dart:71]: Forgot about it. I'll apply this. Thanks

**Review by Bindu-Maharudrappa — DISMISSED** [review]

**Review by kdrob-dolby — APPROVED** [review]

**Pull request merged**